src/client/components/preview-string-filter-menu/preview-string-filter-menu.tsx

```diff
@@ -0,0 +1,258 @@
+  }
+
+  componentDidMount() {
+    this.mounted = true;
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  componentWillReceiveProps(nextProps: PreviewStringFilterMenuProps) {
+    var { searchText } = this.props;
+    const { fetchQueued, loading, dataset } = this.state;
+    // If the user is just typing in more and there are already < TOP_N results then there is nothing to do
+    if (nextProps.searchText.indexOf(searchText) !== -1 && !fetchQueued && !loading && dataset && dataset.data.length < TOP_N) {
+      return;
+    } else {
+      this.setState({
+        fetchQueued: true
+      });
+      this.collectTriggerSearch();
+    }
+  }
+
+  globalKeyDownListener(e: KeyboardEvent) {
+    if (enterKey(e)) {
+      this.onOkClick();
+    }
+  }
+
+  constructFilter(): Filter {
+    var { dimension, filterMode, onClauseChange, searchText } = this.props;
+    var { expression } = dimension;
+
+    var clause: FilterClause = null;
+    if (filterMode === Filter.MATCH && searchText) {
+      clause = new FilterClause({
+        expression,
+        selection: searchText,
+        action: 'match'
+      });
+    }
+
+    return onClauseChange(clause);
+  }
+
+  onOkClick() {
+    if (!this.actionEnabled()) return;
+    var { clicker, onClose } = this.props;
+    var { colors } = this.state;
+    clicker.changeFilter(this.constructFilter(), colors);
+    onClose();
+  }
+
+  onCancelClick() {
+    var { onClose } = this.props;
+    onClose();
+  }
+
+  actionEnabled() {
+    var { essence } = this.props;
+    return !essence.filter.equals(this.constructFilter());
+  }
+
+  renderRows() {
+    var { loading, dataset, fetchQueued  } = this.state;
+    var { dimension, searchText } = this.props;
+
+    var rows: Array<JSX.Element> = [];
+    if (dataset) {
+      var rowStrings = dataset.data.slice(0, TOP_N).map((d) => d[dimension.name]);
+
+      if (searchText) {
+        rowStrings = rowStrings.filter((d) => {
+          try {
+            var escaped = searchText.replace(/\\[^\\]]/g, '\\\\');
+            return new RegExp(escaped, 'g').test(String(d));
+          } catch (e) {
+            return false;
+          }
+        });
+      }
+
+      rows = rowStrings.map((segmentValue) => {
+        var segmentValueStr = String(segmentValue);
+        var match = segmentValueStr.match(searchText);
+        var highlightText = (searchText && match) ? match.join("") : "";
+        return <div
+          className="row no-select"
+          key={segmentValueStr}
+          title={segmentValueStr}
+        >
+          <div className="row-wrapper">
+            <HighlightString className="label" text={segmentValueStr} highlightText={highlightText}/>
+          </div>
+        </div>;
+      });
+    }
+
+    var noResultsMsg: JSX.Element = null;
+    if (!loading && dataset && !fetchQueued && searchText && !rows.length) {
+      noResultsMsg = <div className="message">{'No results for "' + searchText + '"'}</div>;
```

Comment:
maybe put `No results for "%s"` in STRINGS and then do something like `STRINGS.noResults.replace(/%s/g, searchText)`